### PR TITLE
Fix gpt_bigcode/starcoderbase-3b accuracy issue

### PR DESCRIPTION
**Pull request by schoi-habana**
The remove_kv_cache_from_output() function causes an accuracy issue for models that don't perform in-place updates to their KV cache. That's because the function assumes that batch.past_key_values is updated in-place within the model, which isn't the case for Starcoder. 

This change make Starcoder bypass the function and return the kv cache as an output of forward().

### CODE REVIEW

File: server/text_generation_server/models/causal_lm.py [hunk 1]
```diff
@@ -994,14 +997,24 @@ def generate_token(
...
+                )
+            else:
+                batch.logits = self.forward(
+                    input_ids,
+                    batch.attention_mask,
+                    batch.position_ids,
+                    token_idx,
+                    batch.past_key_values,
+                    bypass_hpu_graph=prefill and self.limit_hpu_graph if self.enable_hpu_graph else None,
+                )
```
Comment (tthakkal) [inline server/text_generation_server/models/causal_lm.py]:
```suggestion
            logits = self.forward(
                    input_ids,
                    batch.attention_mask,
                    batch.position_ids,
                    token_idx,
                    batch.past_key_values,
                    bypass_hpu_graph=prefill and self.limit_hpu_graph if self.enable_hpu_graph else None,
                )
            if self.model.config.model_type in ["gpt_bigcode"]:
                batch.logits, batch.past = logits
            else:
                batch.logits = logits
```

Comment (schoi-habana) [reply server/text_generation_server/models/causal_lm.py]: thanks, this looks better. updated with your suggestion

File: server/text_generation_server/models/causal_lm.py [hunk 2]
```diff
@@ -994,14 +997,18 @@ def generate_token(
...
-                bypass_hpu_graph=prefill and self.limit_hpu_graph if self.enable_hpu_graph else None,
-            )
+            logits = self.forward(
+                    input_ids,
+                    batch.attention_mask,
+                    batch.position_ids,
+                    token_idx,
+                    batch.past_key_values,
+                    bypass_hpu_graph=prefill and self.limit_hpu_graph if self.enable_hpu_graph else None,
+                )
```
Comment (tthakkal) [inline server/text_generation_server/models/causal_lm.py]: just fixed indentation
```suggestion
            logits = self.forward(
                input_ids,
                batch.attention_mask,
                batch.position_ids,
                token_idx,
                batch.past_key_values,
                bypass_hpu_graph=prefill and self.limit_hpu_graph if self.enable_hpu_graph else None,
            )
```

**Review by tthakkal — APPROVED** [review]

**Review by mandy-li — APPROVED** [review]
LGTM

**Comment by mandy-li** [comment]
@regisss , pls review and merge if ok for you. thanks

**Review by regisss — APPROVED** [review]
